tree/ntuple/v7/inc/ROOT/RColumnElement.hxx

```diff
@@ -648,6 +648,39 @@ public:
    }
 };
 
+template <>
+class RColumnElement<double, EColumnType::kReal16> : public RColumnElementBase {
+public:
+   static constexpr bool kIsMappable = false;
+   static constexpr std::size_t kSize = sizeof(double);
+   static constexpr std::size_t kBitsOnStorage = 16;
+   RColumnElement() : RColumnElementBase(kSize, kBitsOnStorage) {}
+   bool IsMappable() const final { return kIsMappable; }
+
+   void Pack(void *dst, void *src, std::size_t count) const final
+   {
+      double *doubleArray = reinterpret_cast<double *>(src);
+      std::uint16_t *uint16Array = reinterpret_cast<std::uint16_t *>(dst);
+
+      for (std::size_t i = 0; i < count; ++i) {
+         uint16Array[i] = FloatToHalf(static_cast<float>(doubleArray[i]));
+         ByteSwapIfNecessary(uint16Array[i]);
+      }
+   }
+
+   void Unpack(void *dst, void *src, std::size_t count) const final
+   {
+      double *doubleArray = reinterpret_cast<double *>(dst);
+      std::uint16_t *uint16Array = reinterpret_cast<std::uint16_t *>(src);
+
+      for (std::size_t i = 0; i < count; ++i) {
+         std::uint16_t val = uint16Array[i];
+         ByteSwapIfNecessary(val);
+         doubleArray[i] = HalfToFloat(val);
```

Comment:
This line will widen from single-precision `float` to `double`, right? Maybe add an explicit `static_cast<double>` to make it clear (even though not strictly needed)